### ISSUE TITLE
Mysterious high cpu utilization over time with multiple topics subscription

### ISSUE THREAD

**Issue by jsfwa** [zio/zio-kafka#238]
Hello, we were trying to migrate one of our services from alpakka to zio and discovered strange behavior
So we have couple topics with multiple partitions(about 2-3k rps per partition) and here is our simplified consumer:
```
  Consumer
    .subscribeAnd(Topics(topics))
    .partitionedStream(Serde.long.asTry, Serde.string)
    .mapMParUnordered(50) {
      case (_, s) =>
        s.bufferSliding(1)
          .mapM(_.offset.commitOrRetry(Schedule.recurs(10)))
          .runDrain
    }
    .runDrain
```
After we run it, we got the next picture:
<img width="1707" alt="image" src="https://user-images.githubusercontent.com/7480075/98393794-d9f0ef00-206a-11eb-95b6-61c23903a244.png">

In visualvm cpu utilization looks like this, with spikes every ~2 seconds and slowly increasing avg
<img width="615" alt="image" src="https://user-images.githubusercontent.com/7480075/98396211-6f41b280-206e-11eb-8505-22b6b6feb523.png">

The most time spend on:
<img width="680" alt="image" src="https://user-images.githubusercontent.com/7480075/98396315-9b5d3380-206e-11eb-909a-24687fe8a65a.png">

Its hard to identify real problem, because memory is not leaking, but looks like processors that populates different partitionedStreams at some point start to affect each other until we reset it or do consumer rebalance and the story repeats

Really don't know what else to provide regarding this issue, we have absolutely the same code on akka with same consumer settings and its working smoothly, unfortunately. 

We tried different scenarios:
- Single subscription per topic per application - works fine
- Multiple separate consumers per topic per application - works fine
- One consumer with multiple __slow__ topics/partitions - works fine
- One consumer with multiple topics, with at least one partition with high rps - zio going crazy on cpu
- Also tried to provide different contexts for each substream, change fetch time or batch size - didnt help

Fighting with this issue second week and out of ideas already, please help
zio = 1.0.3
zio-kafka = 0.13.0

Consumer settings
```
      receive.buffer.bytes = 2048576
      send.buffer.bytes = 2048576
      max.partition.fetch.bytes = 2048576
      max.poll.records = 1000
      fetch.max.wait.ms = 10
```
Avg record size =  0.5-1kb before compression

**Comment by iravid** [comment]
Thanks for opening the issue and providing the details. Will look into it this week.

**Comment by aartigao** [comment]
We see a similar pattern in our environments regarding CPU, but also regarding the lag between Kafka record and time were record actually reaches application code. Seems to be related with degradation over time described here (in our case we see that machines start to degrade after 2 days). @iravid do you know if there's a fix? I see that [#4476](https://github.com/zio/zio/pull/4476) is not merged yet. 

We're evaluating a quick fix for periodically restart the streams.

**Comment by aartigao** [comment]
@jsfwa do you know how all the stuff and your investigation ended? did [#4490](https://github.com/zio/zio/pull/4490) fix the problem?

**Comment by guizmaii** [comment]
@jsfwa Any news? 

We've made a lot of improvements lately. Some users reported good performances gain with the `v2.2`.
Did you see them too? Do you still experience this issue or can we close this ticket?

**Comment by erikvanoosten** [comment]
> The most time spend on:
> ![afbeelding](https://github.com/zio/zio-kafka/assets/630600/f2913eda-f7fd-4475-ae12-f498c9397c9b)

This performance problem has been fixed in ZIO 2.0.5 in December 2022 (https://github.com/zio/zio/pull/7557). Please upgrade zio-kafka and zio.

Also, please re-open this issue when you still see performance problems!
